components/form/Form.tsx

```diff
@@ -52,7 +52,7 @@ export type ValidationRule = {
   /** transform a value before validation */
   transform?: (value: any) => any;
   /** custom validate function (Note: callback must be called) */
-  validator?: (rule: any, value: any, callback: any) => any;
+  validator?: (rule: any, value: any, callback: any, source?: any, options?: any) => any;
```

Comment:
Actually... This means that if you want to get `options` from arguments, you must get `source` from arguments at the same time:

```tsx
validator?: (rule: any, value: any, callback: any, source?: any, options?: any) => any;
```

I think that you make a mistake. It is that `async-validator` pass arguments to `validator` and developers get what they want from arguments, not developers pass arguments to `validator`.